### PR TITLE
deps(axe-core): upgrade to 4.1.2

### PR DESCRIPTION
**Pull request by connorjclark**
1) rebaseline the accessibility sample artifact
2) upgrade to 4.1.2

~Do not squash.~

### CODE REVIEW

File: lighthouse-core/test/results/artifacts/artifacts.json
Comment (connorjclark) [inline lighthouse-core/test/results/artifacts/artifacts.json:1761]: These urls are random for each run.

Still glad to have done the rebaseline by itself, otherwise it wouldn't be clear that there is no difference in this version from 4.1.1.

File: lighthouse-core/test/results/artifacts/artifacts.json
Comment (connorjclark) [inline lighthouse-core/test/results/artifacts/artifacts.json:1545]: I'm thinking this went away because the smoke test page has since been "fixed" of a color contrast issue. Or maybe the axe violation check was modified since last time this ran.

Do we want to resolve this?

File: lighthouse-core/test/results/sample_v2.json
Comment (connorjclark) [inline lighthouse-core/test/results/sample_v2.json:2646]: It seems these were trivially passing w/ score 1 because `aria-progressbar-name` check was never found in the axe results, so when LH sees that it just spits out `score: 1`.


side note: https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/audits/accessibility/axe-audit.js#L37 why is score 1 if this is marked n/a? Shouldn't we do `null`, because that's what the audit result will end up being anyway?

File: lighthouse-core/test/results/artifacts/artifacts.json
Comment (patrickhulce) [reply lighthouse-core/test/results/artifacts/artifacts.json:1545]: Given it was a consequence of us rebaselining and not of this version bump, I'm not concerned. IMO, the goal here of sample artifacts is to test at least *some* of the axe-audits and it's not particularly important which ones get exercised.

File: lighthouse-core/test/results/sample_v2.json
Comment (patrickhulce) [reply lighthouse-core/test/results/sample_v2.json:2646]: Extra explanation: I scratched my head for a second, but in order for us to mark as notApplicable if the audit isn't "informative" we need to explicitly find it in axe's `inapplicable` set. 

> why is score 1 if this is marked n/a? Shouldn't we do null, because that's what the audit result will end up being anyway?

Yeah we should. My guess is that while authoring "not applicable" feels closest to "this is passing, don't worry about it" and then 1 gets used.

File: lighthouse-core/gather/gatherers/accessibility.js
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/accessibility.js:25]: @dylanb is this correct?

**Review by patrickhulce — APPROVED** [review]
LGTM though we're both operating on the same source of advice here :)

File: lighthouse-core/gather/gatherers/accessibility.js
Comment (dylanb) [reply lighthouse-core/gather/gatherers/accessibility.js:25]: LGTM

File: lighthouse-core/test/results/artifacts/artifacts.json
Comment (brendankenny) [reply lighthouse-core/test/results/artifacts/artifacts.json:1545]: it is still failing on stable but not in Canary. Looks like the root cause is the text color styling via `/deep/` but that not being supported anymore:

https://github.com/GoogleChrome/lighthouse/blob/c6e14d7c8629c17284c1f773dd6bc8bc61221a47/lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html#L109-L111

Comment (patrickhulce) [reply lighthouse-core/test/results/artifacts/artifacts.json:1545]: Nice find! Do you agree it's not a blocker here @brendankenny ?

Comment (brendankenny) [reply lighthouse-core/test/results/artifacts/artifacts.json:1545]: Yes. We should probably update at some point just because that `/deep/` does nothing (I think we also previously removed the deprecation notice it used to generate because Chrome doesn't even bother anymore), but agreed that's more a smoke test thing, not the purpose of the sample artifacts/sample_v2

File: lighthouse-core/gather/gatherers/accessibility.js
Comment (paulirish) [inline lighthouse-core/gather/gatherers/accessibility.js:70]: 👍   carrying this uid to the url seemed odd. good call.

File: lighthouse-core/gather/gatherers/accessibility.js
Comment (paulirish) [inline lighthouse-core/gather/gatherers/accessibility.js:30]:
```suggestion
    // @ts-ignore axe types don't yet include this new field
```

File: lighthouse-core/gather/gatherers/accessibility.js
Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/accessibility.js:70]: do we use `helpUrl` anywhere?

File: lighthouse-core/gather/gatherers/accessibility.js
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/accessibility.js:99]: why is this needed now? if something's inapplicable, it shouldn't have any nodes at all or it would be applicable...

File: lighthouse-core/gather/gatherers/accessibility.js
Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/accessibility.js:30]: @ts-expect-error :)

File: lighthouse-core/gather/gatherers/accessibility.js
Comment (brendankenny) [inline lighthouse-core/gather/gatherers/accessibility.js:31]: I guess this is fine since `Accessibility` isn't in our `PublicGathererArtifacts`, but we also ship back our `snippet`s which except for eliding are the same thing verbatim, so we aren't exactly creating Fort Knox over here :)

File: lighthouse-core/gather/gatherers/accessibility.js
Comment (connorjclark) [reply lighthouse-core/gather/gatherers/accessibility.js:70]: seems we don't even use this field but w.e.

File: lighthouse-core/gather/gatherers/accessibility.js
Comment (connorjclark) [reply lighthouse-core/gather/gatherers/accessibility.js:99]: help url change. Didn't want to recreate the looping stuff for just that.

**Comment by brendankenny** [comment]
> Do not squash.

is this a description of what you did or a request?

**Comment by connorjclark** [comment]
Don't want to squash the PR, the first commit is just a rebaseline and the second is the actual upgrade.

**Comment by brendankenny** [comment]
> Don't want to squash the PR, the first commit is just a rebaseline and the second is the actual upgrade.

Is there a reason this change is different than usual? We rebaseline artifacts at the same time as updating gatherers lots of times, often when the artifact is hilariously out of date. My vote is don't open the pandoras box for different merge workflows (changes can always go in separate PRs if it's helpful to separate, and smaller PRs are always a bonus :)

**Comment by connorjclark** [comment]
> > Don't want to squash the PR, the first commit is just a rebaseline and the second is the actual upgrade.
> 
> Is there a reason this change is different than usual? We rebaseline artifacts at the same time as updating gatherers lots of times, often when the artifact is hilariously out of date. My vote is don't open the pandoras box for different merge workflows (changes can always go in separate PRs if it's helpful to separate, and smaller PRs are always a bonus :)

https://github.com/GoogleChrome/lighthouse/pull/12075#discussion_r573980214 

This workflow is equivalent to me doing a PR (upgrade axe) on top of the first PR (rebaseline) and waiting for both to land in sequence, except it's simpler to update this way imo. Reviewer can separate the diffs themselves in the commit selector.

I don't do this often, I know @patrickhulce does sometimes, I've never found it problematic wrt reviewing.

**Comment by brendankenny** [comment]
> This workflow is equivalent to me doing a PR (upgrade axe) on top of the first PR (rebaseline) and waiting for both to land in sequence, except it's simpler to update this way imo. Reviewer can separate the diffs themselves in the commit selector.

And a squashed commit is equivalent to updating axe and running `yarn update:sample-artifacts` afterwards :P We didn't even update the artifacts for 4.1.1, so clearly not _that_ important to us.

Landing separate commits for preserving people's work (e.g. they do the majority of a PR but can't finish it) or preserving git but otherwise keeping it simple seems worth it.

Just my opinion. The PR is R: @patrickhulce.

**Comment by patrickhulce** [comment]
re: squash

- The split commits were very nice for reviewing initially, this to me was the primary benefit of the split
- I mostly agree that I don't see a huge benefit to keeping them separate after landing. I mean who knows maybe there's a world in which we want to come back and check whether a particular artifact change was due to 4.1.2 or 4.1.1? Going into repo settings and re-enabling rebase and merge just for this PR and then turning it off again feels a bit like overkill for this benefit, IMO, but if someone with the power to actually change those settings agrees its worth it, I'm fine with it :)
- I don't really think either way is worth a large amount of discussion. Up to whoever has the power to change those repo settings to be convinced by what's already here if it lands rebased or squashed. Sound good?

**Review by dylanb — APPROVED** [review]

**Pull request merged**
